### DISCUSSION TITLE
Issue with loginAs(user) method inside test.

### DISCUSSION THREAD

**Discussion by moitorrijos** [adonisjs/core#3826] [Help] [answered]
Hi everyone.

I have an issue with the loginAs(user) method inside a test I am running.
I already have the @adonis/auth installed and configured, as well as the auth middleware setup.
Furthermore, I also have @adonis/session installed and setup.

This is the code for the test I am running: 
```
import { test } from '@japa/runner'
import User from 'App/Models/User'

test.group('Users', () => {
  test('Get page 1 of users list', async ({ client }) => {
    const user = await User.find(1)
    if (!user) throw new Error('User not found')
    const response = await client.get('/users/page/1').loginAs(user)
    response.dumpBody()
  })
})
```
This is what I get back in dumped by the response.dumpBody():
```
"body"     => {
  errors: [ { message: 'E_UNAUTHORIZED_ACCESS: Unauthorized access' } ]
}
```
Basically, I am having the same issue Aman had in this video:
https://youtu.be/PE0Jmu8Qqjo?t=3253
But I watched it many times and when he corrected the video was cut out. 

Please help. 
🙏
- Juan

**Comment by Barbapapazes** [comment]
Hello, can you provide a reproduction repo to view this error?

**Comment by moitorrijos** [comment]
Yes, absolutely. This is it:
[https://github.com/moitorrijos/seagulls-api](seagulls-api)

**Comment by thetutlage** [comment, marked as answer]
Make sure to set the `SESSION_DRIVER` to `memory`. Following is a screenshot from docs.

<img width="775" alt="CleanShot 2022-07-28 at 11 50 37@2x" src="https://user-images.githubusercontent.com/1706381/181434654-a0932725-bf12-4a11-a159-2adfeb6418c9.png">

**Reply by moitorrijos** [reply]
Thank you. I don't know how I missed that in the docs.

**Reply by vandsonfalcao** [reply]
Thanks, with this code I discovered how to authenticate in test requests.
